### PR TITLE
Move ESNext to default code example shown

### PR DESCRIPTION
**Pull request by mkaz**
## Description

This changes the documentation in the block tutorials. It adds clarification around ES5 and ESNext and by default shows ESNext code. This sets the documentation for having ESNext as a more preferred syntax, but does not remove any of the ES5 examples.

The reasoning behind this change is that it can be quite confusing to see mostly ES5 code in the documentation but then it is not the primary use case "in the wild". This makes for an odd disconnect, and I think the docs should reflect the real usage.

For example, the core Gutenberg packages and block library are all in ESNext, as well as the readme.md examples for each component is only in ESNext syntax. This is easiest to see using the [WP Storybook](https://wp-storybook.netlify.com/?path=/story/components-basecontrol--basic)

Additionally the majority of plugins and blocks from other code bases are in ESNext syntax, people learning Gutenberg will use these as a reference which can be confusing. [CoBlocks](https://github.com/godaddy/coblocks/tree/master/src/blocks) is just but one example of many.

Initially [discussed in Editor Chat](https://make.wordpress.org/core/2019/10/09/editor-chat-summary-wednesday-9-october-2019/)

## How has this been tested?

Confirm documentation still makes sense and works as described. No real functional change, the majority is just reordering of ESNext code block ahead of ES5.

View [live on the branch here](https://github.com/WordPress/gutenberg/tree/docs/esnext-block-tutorial/docs/designers-developers/developers/tutorials/block-tutorial)

## Types of changes

Documentation updates.

### CODE REVIEW

File: docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md [hunk 1]
```diff
@@ -5,6 +5,28 @@ In the previous step, the block had applied its own styles by an inline `style`
 The editor will automatically generate a class name for each block type to simplify styling. It can be accessed from the object argument passed to the edit and save functions. In step 2, we will create a stylesheet to use that class name.
 
 {% codetabs %}
+{% ESNext %}
+```js
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/block-tutorial/applying-styles-with-stylesheets.md]:
```suggestion
```jsx
```

File: docs/designers-developers/developers/tutorials/block-tutorial/block-controls-toolbar-and-sidebar.md [hunk 2]
```diff
@@ -11,6 +11,85 @@ When the user selects a block, a number of control buttons may be shown in a too
 You can also customize the toolbar to include controls specific to your block type. If the return value of your block type's `edit` function includes a `BlockControls` element, those controls will be shown in the selected block's toolbar.
 
 {% codetabs %}
+{% ESNext %}
+```js
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/block-tutorial/block-controls-toolbar-and-sidebar.md]:
```suggestion
```jsx
```

File: docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md [hunk 3]
```diff
@@ -18,6 +18,38 @@ Block attributes can be used for any content or setting you want to save for tha
 The following code example shows how to create a dynamic block that shows only the last post as a link.
 
 {% codetabs %}
+{% ESNext %}
+```js
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md]:
```suggestion
```jsx
```

File: docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md [hunk 4]
```diff
@@ -150,6 +154,26 @@ Gutenberg 2.8 added the [`<ServerSideRender>`](/packages/components/src/server-s
 *Server-side render is meant as a fallback; client-side rendering in JavaScript is always preferred (client rendering is faster and allows better editor manipulation).*
 
 {% codetabs %}
+{% ESNext %}
+```js
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md]:
```suggestion
```jsx
```

File: docs/designers-developers/developers/tutorials/block-tutorial/introducing-attributes-and-editable-fields.md [hunk 5]
```diff
@@ -54,6 +53,46 @@ Because `RichText` allows for nested nodes, you'll most often use it in conjunct
 Here is the complete block definition for Example 03.
 
 {% codetabs %}
+{% ESNext %}
+```js
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/block-tutorial/introducing-attributes-and-editable-fields.md]:
```suggestion
```jsx
```

File: docs/designers-developers/developers/tutorials/block-tutorial/writing-your-first-block-type.md [hunk 6]
```diff
@@ -6,40 +6,71 @@ Blocks containing static content are implemented entirely in JavaScript using th
...
 
-If you were to use a component from the `wp-editor` package, for example the RichText component, you would also need to add `wp-editor` to the dependency list.
 
 ## Registering the Block
 
 With the script enqueued, let's look at the implementation of the block itself:
 
 {% codetabs %}
+{% ESNext %}
+```js
```
Comment (mkaz) [inline docs/designers-developers/developers/tutorials/block-tutorial/writing-your-first-block-type.md]:
```suggestion
```jsx
```

**Review by mkaz — COMMENTED** [review]
Switch JS to JSX for code syntax

File: docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md [hunk 7]
```diff
@@ -120,20 +120,24 @@ function gutenberg_examples_05_dynamic_render_callback( $attributes, $content )
 	);
 }
 
-function gutenberg_examples_05_dynamic() {
+function gutenberg_examples_dynamic() {
+	// automatically load dependencies and version
+	$asset_file = include( plugin_dir_path( __FILE__ ) . 'build/index.asset.php');
+
 	wp_register_script(
 		'gutenberg-examples-05',
```
Comment (nosolosw) [inline docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md]:
```suggestion
		'gutenberg-examples-dynamic',
```

File: docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md [hunk 8]
```diff
@@ -120,20 +120,24 @@ function gutenberg_examples_05_dynamic_render_callback( $attributes, $content )
...
+		plugins_url( 'build/block.js', __FILE__ ),
+		$asset_file['dependencies'],
+		$asset_file['version']
 	);
 
-	register_block_type( 'gutenberg-examples/example-05-dynamic', array(
-		'editor_script' => 'gutenberg-examples-05',
-		'render_callback' => 'gutenberg_examples_05_dynamic_render_callback'
+	register_block_type( 'gutenberg-examples/example-dynamic', array(
+		'editor_script' => 'gutenberg-examples',
```
Comment (nosolosw) [inline docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md]:
```suggestion
		'editor_script' => 'gutenberg-examples-dynamic',
```

**Review by oandregal — APPROVED** [review]
I've left a couple of suggestions. Other than that, this is ready to land.

**Pull request merged**